cdap-security/src/main/java/co/cask/cdap/security/impersonation/SecurityUtil.java

```diff
@@ -278,4 +279,28 @@ public static void verifyOwnerPrincipal(NamespacedEntityId existingEntity,
                                                     Constants.Security.PRINCIPAL));
     }
   }
+
+  /**
+   * Helper function to get the effective owner of an entity. It will check the owner store to get the namespace
+   * owner if the provided owner principal is null.
+   *
+   * Note that this function should only be used when we
```

Comment:
`Note that this method need not be used after the entity is created, in that case simply use {@link OwnerAdmin.getImpersonationPrincipal()}`